### PR TITLE
Add CIFAR-10/100 convnet example

### PR DESCRIPTION
**Pull request by bkvogel**
Chainer already has dataset loaders for CIFAR-10 and CIFAR-100 but no corresponding examples. This PR adds an example to train a convolutional neural network on either of these datasets.

### CODE REVIEW

File: examples/cifar/models/VGG.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            block3_3=Block(256, 3),
+            block4_1=Block(512, 3),
+            block4_2=Block(512, 3),
+            block4_3=Block(512, 3),
+            block5_1=Block(512, 3),
+            block5_2=Block(512, 3),
+            block5_3=Block(512, 3),
+            fc1=L.Linear(None, 512, nobias=True),
+            bn_fc1=L.BatchNormalization(512),
+            fc2=L.Linear(None, class_labels, nobias=True)
```
Comment (rezoo) [inline examples/cifar/models/VGG.py]: It seems that nn.Linear() in torch includes bias term by default. Why ignoring bias terms in your models?

File: examples/cifar/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Convolutional nueral networks for CIFAR-10 and CIFAR-100 Classification
```
Comment (rezoo) [inline examples/cifar/README.md]: s/nueral/neural/

Comment (bkvogel) [inline examples/cifar/README.md]: I will fix.

File: examples/cifar/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+This is a an example of a convolutional neural network (convnet) applied to an image classification task using the CIFAR-10 or CIFAR-100 dataset. The CIFAR datasets can be a good choice for initial experiments with convnets because the size and number of images is small enough to allow typical models to be trained in a reasonable amount of time. However, the classification task is still challenging because natural images are used.
```
Comment (bkvogel) [inline examples/cifar/README.md]: I will fix.

File: examples/cifar/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+For example, to run the default model, which uses CIFAR-10 and GPU 0:
+```
+train_cifar.py
+```
+
+to run the CIFAR-100 dataset on GPU 1:
+```
+train_cifar.py --gpu=1 --dataset='cifar100'
+```
```
Comment (bkvogel) [inline examples/cifar/README.md]: I agree. I will mention.

File: examples/cifar/models/VGG.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        )
+        self.train = True
+
+    def __call__(self, x):
+        # 64 channel blocks:
+        h = self.block1_1(x, self.train)
+        h = F.dropout(h, ratio=0.3, train=self.train)
+        h = self.block1_2(h, self.train)
+        h = F.max_pooling_2d(h, 2, 2)
```
Comment (bkvogel) [inline examples/cifar/models/VGG.py]: I agree and will make the change.

File: examples/cifar/models/VGG.py (hunk 1 above)
Comment (rezoo) [inline examples/cifar/models/VGG.py]: I see. Thanks.

**Comment by rezoo** [comment]
Good. Then I would like to merge this one.

**Comment by bkvogel** [comment]
Thanks for your review. Did you get my latest commit to fix another pep8 style error before merging? Otherwise, it will cause a build error.

**Comment by rezoo** [comment]
At least in my environment, pep8 and flake8 do not say any style errors.

**Comment by rezoo** [comment]
```
$ autopep8 models/VGG.py 
--- original/models/VGG.py
+++ fixed/models/VGG.py
@@ -20,6 +20,7 @@
         pad (int): The padding to use for the convolution.
 
     """
+
     def __init__(self, out_channels, ksize, pad=1):
         super(Block, self).__init__(
             conv=L.Convolution2D(None, out_channels, ksize, pad=pad,
@@ -55,6 +56,7 @@
         class_labels (int): The number of class labels.
 
     """
+
     def __init__(self, class_labels=10):
         super(VGG, self).__init__(
             block1_1=Block(64, 3),
```
Ah I see, pep8 and flake8 are OK, but autopep8 says warnings.

**Comment by bkvogel** [comment]
I made a simple PR to fix this: #1982
Could you review? Thanks!

**Pull request merged**